sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/specialized/ShareLeaseClient.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Acquires a lease for write and delete operations. All leases are infinite.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseClient.acquireLeaseWithResponse#Duration-Context}
+     *
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return The lease ID.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<String> acquireLeaseWithResponse(Duration timeout, Context context) {
+        return StorageImplUtils.blockWithOptionalTimeout(this.client.acquireLeaseWithResponse(context), timeout);
+    }
+
+    /**
+     * Releases the previously acquired lease.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseClient.releaseLease}
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void releaseLease() {
+        releaseLeaseWithResponse(null, Context.NONE);
+    }
+
+    /**
+     * Releases the previously acquired lease.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseClient.releaseLeaseWithResponse#Duration-Context}
+     *
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return A response containing status code and HTTP headers.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> releaseLeaseWithResponse(Duration timeout, Context context) {
+        return StorageImplUtils.blockWithOptionalTimeout(this.client.releaseLeaseWithResponse(context), timeout);
+    }
+
+    /**
+     * Breaks the previously acquired lease, if it exists. Leases will break immediately.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseClient.breakLease}
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void breakLease() {
+        breakLeaseWithResponse(null, Context.NONE).getValue();
+    }
+
+    /**
+     * Breaks the previously acquired lease, if it exists. Leases will break immediately.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.specialized.ShareLeaseClient.breakLeaseWithResponse#Duration-Context}
+     *
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return The remaining time in the broken lease in seconds.
```

Comment:
Same comment as the async client about leases being infinite and breaking immediately.